bundler/lib/bundler/templates/newgem/github/workflows/main.yml.tt

```diff
@@ -10,7 +18,7 @@ jobs:
     - name: Set up Ruby
       uses: ruby/setup-ruby@v1
       with:
-        ruby-version: <%= RUBY_VERSION %>
+        ruby-version: ${{ matrix.ruby }}
```

Comment:
I would prefer to keep `RUBY_VERSION` in here, since I think there is no need for any matrix to support 1 particular version only. Alternatively (if we would like to keep matrix) it would be enough to don't hardcode `3.0.0` in the template and use `RUBY_VERSION` in the matrix instead.

Reply:
@simi , I'll probably add other versions. I asked @deivid-rodriguez, if we'll take this [comment](https://github.com/rubygems/rubygems/issues/4230#issue-778568037) into account or not. Would appreciate if you can give your thoughts on that as well. You're totally right for the hardcoded version, will update it.

Reply:
You're thinking something like the following?
```yaml
strategy:
  matrix:
    ruby: [ <%= RUBY_VERSION %> ]
```

At least this leaves the template such that the matrix could be expanded.

Since most gems support a range of rubies,  one could...

Reply:
@MSP-Greg , yep something like that with other versions in it: current, the two preceding ones and ruby-head

Reply:
yes, the rationale for moving to a matrix is that it makes it easier to add new version, but on the related issue @marcandre also suggests adding `ruby-head`, and also a couple more stable rubies. In my opinion it's a good idea. @simi What do you think?

Reply:
Ok, let's start with a matrix including the current ruby and `ruby-head` I guess.

Reply:
Actually `ruby-head` should go into `allowed_failures` by default, and the configuration might get tricky... Maybe let's leave just `RUBY_VERSION` for now and discuss how to add `ruby-head` separately?